src/utils/operator.ts

```diff
@@ -69,30 +69,73 @@ export function getOrderFilledAmount(order: RawOrder): { amount: BigNumber; perc
   return { amount: executedAmount, percentage: executedAmount.div(totalAmount) }
 }
 
-export function getSurplus(
-  inputAmount: BigNumber | string,
-  executedAmount: BigNumber | string,
-): { amount: BigNumber; percentage: BigNumber } {
-  // Just as a nicety, allow both input as strings
-  // inputAmount has ne need for conversion since BigNumber takes care of it when used as a parameter
-  // executedAmount needs to be converted though
-  const _executedAmount = typeof executedAmount === 'string' ? new BigNumber(executedAmount) : executedAmount
-
-  const amount = _executedAmount.gt(inputAmount) ? _executedAmount.minus(inputAmount) : ZERO_BIG_NUMBER
-  const percentage = amount.dividedBy(inputAmount)
+type Surplus = {
+  amount: BigNumber
+  percentage: BigNumber
+}
+
+type BigNumberIsh = string | BigNumber
+
+/**
+ * Calculates SELL surplus based on buy amounts
+ *
+ * @param buyAmount buyAmount
+ * @param executedBuyAmount executedBuyAmount
+ * @returns Sell surplus
+ */
+export function getSellSurplus(buyAmount: BigNumberIsh, executedBuyAmount: BigNumberIsh): Surplus {
+  const _buyAmount = new BigNumber(buyAmount)
+  const _executedAmount = new BigNumber(executedBuyAmount)
+  // SELL order has the sell amount fixed (minus the fees),
+  // so it'll buy AT LEAST `buyAmount`
+  // Surplus is in the form of additional buy amount, buying more than `buyAmount`
+  // Since this is a `fillOrKill`, whenever `executedBuyAmount >= buyAmount` the order was fully executed.
+  // The difference between `executedBuyAmount - buyAmount` is the surplus.
+  const amount = _executedAmount.gt(_buyAmount) ? _executedAmount.minus(_buyAmount) : ZERO_BIG_NUMBER
+  const percentage = amount.dividedBy(_buyAmount)
 
   return { amount, percentage }
 }
 
-export function getOrderSurplus(order: RawOrder): { amount: BigNumber; percentage: BigNumber } {
-  const { kind, buyAmount, sellAmount } = order
+/**
+ * Calculates BUY surplus based on sell amounts
+ * WARNING: Respective fees must be already discounted from both parameters
+ *
+ * @param sellAmount sellAmount minus feeAmount
+ * @param executedSellAmount executedSellAmount minus executedFeeAmount
+ * @returns Buy surplus
+ */
+export function getBuySurplus(sellAmount: BigNumberIsh, executedSellAmount: BigNumberIsh): Surplus {
+  const _sellAmount = new BigNumber(sellAmount)
+  const _executedAmount = new BigNumber(executedSellAmount)
+  // BUY order has the buy amount fixed, so it'll sell AT MOST `sellAmount` (minus the fees)
+  // Surplus will come in the form of a "discount", selling less than `sellAmount`
+  // Since this is a `fillOrKill`, whenever `executedSellAmount` > 0 the order was fully executed.
```

Comment:
For simplicity, this PR only addresses fillOrKill.
Also, we do not yet have any partial fill orders to display.
We can address that when they are available.

Also also, does it make sense to display the order surplus for partial fills? Or should that be displayed for each individual trade instead?

With those questions in mind, I chose to deal only with fillOrKill for now.
Happy to get to work on it in a follow up PR in case we want to prioritize this right now.

Reply:
I got your point now.
This function should be usable on fillOrKill amounts or trade amounts.
Removed the reference to fillOrKill from comment